Patch-Source: https://code.qt.io/cgit/pyside/pyside-setup.git/commit/?h=6.3&id=4ae3d8f7dc43a14ea2bb90933678d699ce552078
From 703d975f16aff95bc9014a2689a3ae824b5a552f Mon Sep 17 00:00:00 2001
From: Christian Tismer <tismer@stackless.com>
Date: Wed, 20 Apr 2022 14:00:32 +0200
Subject: setup: fix PySide6.__all__ after the wheel split, augmented

The original fix had a criterion to use the unchanged
module list when a build is active. The determination if a
build is in progress was too setup.py specific.

Using a compiled-in variable solved in a general way.

This patch can be removed completely when the wheel
split method is getting aware of __all__ .

Thanks to Antonio Rojas for the nice hint.

Change-Id: Iaf4cb28056b7f80f2f0c9fb5604c3033bfc1aba5
Fixes: PYSIDE-1890
Pick-to: 6.3
Reviewed-by: Friedemann Kleint <Friedemann.Kleint@qt.io>
---
 sources/pyside6/PySide6/__init__.py.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sources/pyside6/PySide6/__init__.py.in b/sources/pyside6/PySide6/__init__.py.in
index d8439985d..8f6c47b87 100644
--- a/sources/pyside6/PySide6/__init__.py.in
+++ b/sources/pyside6/PySide6/__init__.py.in
@@ -123,7 +123,8 @@ def _find_all_qt_modules():
     location = Path(__file__).resolve().parent
 
     # Note: We should _not_ call this function while still building, but use the existing value!
-    in_build = location.parents[1].name == "build"
+    in_build = Path("@CMAKE_BINARY_DIR@") in location.parents
+
     if in_build:
         return __all__
 
-- 
cgit v1.2.1

